sunpy/timeseries/sources/goes.py

```diff
@@ -228,12 +227,12 @@ def _parse_netcdf(filepath):
                 raise ValueError(f"No flux data (either a_flux or xrsa_flux) found in file: {filepath}")
             flux_name_a = flux_name.name
             flux_name_b = flux_name_a.replace("a", "b")
-            xrsa = np.array(h5nc[flux_name_a])
-            xrsb = np.array(h5nc[flux_name_b])
+            xrsa = np.asarray(h5nc[flux_name_a])
```

Comment:
numpy only wants to accept array inputs that implement copy and dtype. So switching to asarray bypasses this warning.